src/Components/Header.js

```diff
@@ -1,13 +1,53 @@
 import React, { Component } from 'react';
+import {Button, Navbar, Nav, NavItem } from 'react-bootstrap';
+import AuthService from '../services';
 
 class Header extends Component {
+  constructor(props) {
+    super(props)
+
+    this.auth = new AuthService()
+  }
   render() {
     return (
       <div className="Header-Div">
-      NavBar Goes Here (:
+        <Navbar>
+          <Navbar.Header>
+            <Navbar.Brand>
+              <a href="/">Insert Name Here</a>
+            </Navbar.Brand>
+          </Navbar.Header>
+          <Nav>
+            <NavItem eventKey={1} href="/">
+              Home
+            </NavItem>
+            {!this.auth.loggedIn() &&
+            <NavItem eventKey={2} href="/user/:id/tasks">
+             My Tasks
+            </NavItem>}
+            {this.auth.loggedIn() &&
+            <NavItem eventKey={3} href='/tasks'>
+              All Tasks
+            </NavItem>}
+            {this.auth.loggedIn() &&
+            <NavItem eventKey={4} href="/">
+              Edit Tasks
+            </NavItem>}
+            {this.auth.loggedIn() &&
+            <NavItem onClick={this.handleClick} eventKey={5} href="/">
+              Logout
+            </NavItem>}
+          </Nav>
+        </Navbar>
       </div>
     );
   }
+
+  handleClick = () => {
+    this.auth.logout();
+    this.props.logout();
```

Comment:
Does this component have a `logout()` function passed in via the props ?  And if so, why ?  It appears `this.auth` can handle it, right ?

Reply:
This still needs to be fixed ^ .  That might be part of another PR ?  Or you could do it in this PR.   By "fixed", I mean getting your user data (state) higher in the react tree.

If you want to do this work in another PR, I think you are ok to close this PR.